ci/test_python_singlegpu.sh

```diff
@@ -4,6 +4,9 @@
 # Common setup steps shared by Python test jobs
 source "$(dirname "$0")/test_python_common.sh"
 
+rapids-mamba-retry install -c conda-forge scikit-learn=1.2
+pip install treelite==3.1
```

Comment:
We should specify these requirements in `dependencies.yaml` (it also supports pip-installed packages). The pip install for treelite would go next to the `git` URL for `dask-glm`.

Reply:
We do not want to have `pip treelite` as a dependency but the regular conda we have, I used it temporarily because I pushed the PR just before the conda 3.1 packages were published.

Reply:
The 3.1 package is now on Conda: https://anaconda.org/conda-forge/treelite